### DISCUSSION TITLE
getServerSideProps redirect with props

### DISCUSSION THREAD

**Discussion by mikey-mo** [vercel/next.js#28273] [Help]
Hi all.

Currently have `getServerSideProps` running with a try / catch and an async call to service:

```
export const getServerSideProps = async () => {
  try {
    const {
      data
    } = await linksServices.getById('some id);

    return {
      props: {
        data,
      },
    };
  } catch (e) {
    return {
      redirect: {
        destination: '/error',
        permanent: true,
      },
    };
  }
};

export default Index;
```

On the catch I want to redirect to an error page based on a specific error.  I have generic error page that takes some copy props / ctas.  I want to be able to redirect AND pass these props, but I cannot figure how to do so.

I could either make a separate page component for this specific error page, or I could set the page prop defaults with the copy / ctas -- neither of these solutions sound great to me.

Hoping there is something I missed that someone can point out.

Thanks.

**Comment by moroshko** [comment]
@timneutkens @leerob Is it possible to pass props to the destination page when redirecting from getServerSideProps?

**Comment by mikey-mo** [comment]
Has anyone had any success w/this outside of having a specific page / query arguments / default props?
